app/controllers/concerns/model_controller_behavior.rb

```diff
@@ -7,11 +7,13 @@ module ModelControllerBehavior
 
   def new
     @form = form_class.new(resource_class.new).prepopulate!
+    authorize! :create, resource_class
     @collections = ::Draper::CollectionDecorator.decorate(query_service.find_all_of_model(model: Collection))
   end
 
   def create
     @form = form_class.new(resource_class.new)
+    authorize! params[:action], @form.model
```

Comment:
I used that because it's all over the CanCanCan docs, but I'm happy to change it.